presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/RemoveAggregationInSemiJoin.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.sql.planner.iterative.rule;
+
+import com.google.common.collect.ImmutableList;
+import io.prestosql.matching.Capture;
+import io.prestosql.matching.Captures;
+import io.prestosql.matching.Pattern;
+import io.prestosql.sql.planner.iterative.Rule;
+import io.prestosql.sql.planner.plan.AggregationNode;
+import io.prestosql.sql.planner.plan.SemiJoinNode;
+
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static io.prestosql.matching.Capture.newCapture;
+import static io.prestosql.sql.planner.plan.Patterns.SemiJoin.getFilteringSource;
+import static io.prestosql.sql.planner.plan.Patterns.aggregation;
+import static io.prestosql.sql.planner.plan.Patterns.semiJoin;
+
+/**
+ * Remove the aggregation node in the Filtering source of a Semi join if it produces distinct rows.
+ */
+public class RemoveAggregationInSemiJoin
```

Comment:
`RemoveAggregationFromFilteringSourceInSemiJoin`?

Reply:
Better name would be `RemoveAggregationFromSemiJoinFilteringSource`

Reply:
I suggested the name he used in an offline conversation. The name doesn’t need to be ultra precise, but should capture the gist of what the rule is about.